upgrade/capsule.py

```diff
@@ -83,35 +83,41 @@ def satellite_capsule_upgrade(cap_host, sat_host):
 
     """
     logger.highlight('\n========== CAPSULE UPGRADE =================\n')
+    if zstream:
+        if not settings.upgrade.from_version == settings.upgrade.to_version:
+            logger.highlight('Z-Stream Capsule Upgrade cannot be performed '
+                             'when FROM and TO versions differ. Aborting...')
+            sys.exit(1)
     # Check the capsule sync before upgrade.
-    logger.info("Check the capsule sync after satellite upgrade to verify sync operation "
-                "with n-1 combination")
-    execute(capsule_sync, cap_host, host=sat_host)
-    wait_untill_capsule_sync(cap_host)
-    setup_capsule_firewall()
+    if not zstream:
+        logger.info("Checking the capsule sync after satellite upgrade to verify sync operation "
+                    "with n-1 combination")
+        execute(capsule_sync, cap_host, host=sat_host)
+        wait_untill_capsule_sync(cap_host)
+        setup_capsule_firewall()
     os_ver = int(settings.upgrade.os.strip('rhel'))
     ak_name = settings.upgrade.capsule_ak[settings.upgrade.os]
     run(f'subscription-manager register '
         f'--org="Default_Organization" --activationkey={ak_name} --force')
     logger.info(f'Activation key {ak_name} enabled all capsule repositories')
     run('subscription-manager repos --list')
-    maintenance_repo = RH_CONTENT['maintenance']['label']
-    capsule_repo = RH_CONTENT['capsule']['label']
     client = 'client' if Version(settings.upgrade.to_version) > Version('6.10') else 'tools'
-    client_repo = RH_CONTENT[client]['label']
-
+    capsule_repos = [
+        RH_CONTENT['capsule']['label'],
+        RH_CONTENT['maintenance']['label'],
+        RH_CONTENT[client]['label']
+    ]
     with fabric_settings(warn_only=True):
-        if os_ver == 7:
-            enable_disable_repo(enable_repos_name=[RH_CONTENT['ansible']['label']])
-        if settings.upgrade.distribution == 'cdn':
-            enable_disable_repo(enable_repos_name=[capsule_repo, maintenance_repo, client_repo])
+        if settings.upgrade.distribution == "cdn":
+            enable_disable_repo(enable_repos_name=capsule_repos)
+            if os_ver == 7:
+                enable_disable_repo(enable_repos_name=[RH_CONTENT['ansible']['label']])
```

Comment:
Wrong place - we need ansible from CDN always !
Put away from `if settings.upgrade.distribution == "cdn"` as it was originally

Reply:
Updated